cpp/src/parquet/file_reader.cc

```diff
@@ -268,6 +272,7 @@ class SerializedRowGroup : public RowGroupReader::Contents {
   std::unique_ptr<RowGroupMetaData> row_group_metadata_;
   ReaderProperties properties_;
   int row_group_ordinal_;
+  std::unordered_set<int>& prebuffered_column_chunks_;
```

Comment:
IMO, this would be better to use `const std::unordered_set<int>`, without reference.